### PR TITLE
🌱 Bump golang 1.22 in release 1.6

### LINKED ISSUE #1829: Bump CAPM3 1.6 release branch to Golang 1.22
Release branch release-1.6 has Golang 1.21.x in use. Golang 1.21 support ends in August 2024, when Golang 1.23 is released.

CAPM3 1.6 is in support for some time still, so we need to bump it to Go 1.22 before next patch release.

### PR DESCRIPTION
**Pull request by NymanRobin**
This is done since golang 1.20 is already deprecated

Fixes #1829

### CODE REVIEW

**Comment by NymanRobin** [comment]
Corresponding projectinfra PR that is needed first is here: https://github.com/metal3-io/project-infra/pull/825

**Review by adilGhaffarDev — COMMENTED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/metal3-io/cluster-api-provider-metal3
 
-go 1.20
+go 1.22
```
Comment (dtantsur) [inline go.mod:3]: Similarly to BMO, go.mod should not be touched by this sort of changes.

Comment (NymanRobin) [reply go.mod]: This is taken care of!

**Review by smoshiur1237 — COMMENTED** [review]
/approve

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Comment by Sunnatillo** [comment]
/retitle 🌱 Bump golang 1.22 in release 1.6

**Pull request merged**